lib/vsc/install/shared_setup.py

```diff
@@ -156,6 +165,88 @@ def remove_extra_bdist_rpm_files(pkgs=None):
 
     return res
 
+class vsc_sdist(sdist):
+    """
+    Upon sdist, add this vsc.install.shared_setup to the sdist
+    and modifed the shipped setup.py to be able to use this
+    """
+    def _mod_setup_py(self, base_dir, external_dir, new_shared_setup):
+        """Modify the setup.py in the distribution directory"""
+
+        # re-copy setup.py, to avoid hardlinks (code from setupstools)
+        dest = os.path.join(base_dir, 'setup.py')
+        log.error('recopying dest %s if hardlinked' % dest)
+        if hasattr(os, 'link') and os.path.exists(dest):
+            # unlink and re-copy, since it might be hard-linked, and
+            # we don't want to change the source version
+            os.unlink(dest)
+            self.copy_file('setup.py', dest)
+
+        fh = open(dest, 'r')
+        code = fh.read()
+        fh.close()
+
+        # look for first line that does someting with vsc.install and shared_setup
+        reg = re.search(r'^.*vsc.install.*shared_setup.*$', code, re.M)
+        if not reg:
+            raise Exception("No vsc.install shared_setup in setup.py?")
+
+        # insert sys.path hack
+        before = reg.start()
+        # no indentation
+        code = code[:before] + NEW_SHARED_SETUP_HEADER_TEMPLATE % (new_shared_setup, VERSION, external_dir) + code[before:]
+
+        # replace 'vsc.install.shared_setup' -> new_shared_setup
+        code = re.sub(r'vsc\.install\.shared_setup', new_shared_setup, code)
+        # replace 'from vsc.install import shared_setup' -> import new_shared_setup as shared_setup
+        code = re.sub(r'from\s+vsc.install\s+import\s+shared_setup', 'import %s as shared_setup', code)
+
+        # write it
+        fh = open(dest, 'w')
+        fh.write(code)
+        fh.close()
+
+    def _add_shared_setup(self, base_dir, external_dir, new_shared_setup):
+        """Create the new shared_setup in distribution directory"""
+
+        ext_dir = os.path.join(base_dir, external_dir)
+        os.mkdir(ext_dir)
+
+        dest = os.path.join(ext_dir, '%s.py' % new_shared_setup)
+        log.error('inserting shared_setup as %s' % dest)
+        try:
+            source_code = inspect.getsource(sys.modules[__name__])
+        except:
+            raise Exception("sdist requires access shared_setup source")
+        try:
+            fh = open(dest, 'w')
+            fh.write(source_code)
+            fh.close()
+        except:
```

Comment:
`... as err`, and mention what went wrong in the `raise` below? don't hide stuff...